### PR TITLE
[12.0][MIG] stock_available_immediately missing module needed by stock_available module

### PR DESCRIPTION
**Pull request by kosz85**
Migrated missing module that is required by stock_available module to be functional.

### CODE REVIEW

**Comment by rousseldenis** [comment]
@kosz85 Thanks for your PR.

Could you update the title of this PR ? Maybe something like [12.0][MIG] stock_available_immediately as this module is not 'needed' by stock_available.

Could you also integrates commit history from past branches ?

See https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-12.0

Thanks

**Comment by kosz85** [comment]
This module IS needed, as without it it makes just no sense, I will take a look into link you provided, and fix it, after signing that license.

**Comment by rousseldenis** [comment]
> This module IS needed,

@kosz85 Ok, in a functional point of view. But technically, this one depends on the other :smile:

File: stock_available_immediately/__init__.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# -*- coding: utf-8 -*-
```
Comment (rousseldenis) [inline stock_available_immediately/__init__.py:1]: Please remove all utf-8 lines as not needed anymore with python 3

File: stock_available_immediately/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# -*- coding: utf-8 -*-
+# Copyright 2014 Camptocamp, Akretion, Numérigraphe
+# Copyright 2016 Sodexis
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+{
+    "name": "Ignore planned receptions in quantity available to promise",
+    "version": "10.0.1.0.0",
```
Comment (rousseldenis) [inline stock_available_immediately/__manifest__.py:8]: Could you please change module version to '12.0.1.0.0' ?

File: stock_available_immediately/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    "depends": ["stock_available"],
+    "author": "Camptocamp,Sodexis,Odoo Community Association (OCA)",
+    "license": "AGPL-3",
+    "category": "Hidden",
```
Comment (rousseldenis) [inline stock_available_immediately/__manifest__.py:12]: Could you maybe add some missing entries like in https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py ?

Could you change category to 'Warehouse Management' as we can find this module more easily on OCA shop ?

File: stock_available_immediately/tests/test_stock_available_immediately.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    def test01_stock_levels(self):
+        """checking that immediately_usable_qty actually reflects \
+the variations in stock, both on product and template"""
+        moveObj = self.env['stock.move']
+        productObj = self.env['product.product']
+        templateObj = self.env['product.template']
+        supplier_location = self.env.ref('stock.stock_location_suppliers')
+        stock_location = self.env.ref('stock.stock_location_stock')
+        customer_location = self.env.ref('stock.stock_location_customers')
+        uom_unit = self.env.ref('product.product_uom_unit')
```
Comment (rousseldenis) [inline stock_available_immediately/tests/test_stock_available_immediately.py:20]: uom's are managed in v12 in 'uom' module.

```suggestion
        uom_unit = self.env.ref('uom.product_uom_unit')
```

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review.

**Comment by rousseldenis** [comment]
@kosz85

**Comment by kosz85** [comment]
Sorry In the end, I won't do that. I do not have CLA signed, and it's a bother to sign it.  This is just code checkouted from previous releases (10.0 if i recall correctly) where it was existing, and it's working, so in fact anyone can fix it, and can use it.
Our company decided to use our own solution, so we won't use this one.

**Comment by rousseldenis** [comment]
@kosz85 Ok. Don't hesitate to come back!

**Comment by rousseldenis** [comment]
> and it's a bother to sign it.

You can read https://odoo-community.org/page/cla for further information. If you have questions, don't hesitate.

**Pull request closed without merging**